src/GherkinCompatibilityMode.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/*
+ * This file is part of the Behat Gherkin Parser.
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Behat\Gherkin;
+
+enum GherkinCompatibilityMode: string
+{
+    case LEGACY = 'legacy';
+
+    case GHERKIN_32 = 'gherkin-32';
```

Comment:
I have tied this to a gherkin version, because we may in future want to use similar logic to support new gherkin majors.

The downside is that unless we solve all the cucumber parity issues in the first release, the behaviour in `gherkin-32` mode will change in subsequent releases. That may be acceptable if we advertise to users that the `gherkin-32` mode is expected to be unstable for a few releases - they can always stick with the default legacy mode.

I don't want to expose individual feature flags to end users as that will hugely complicate our ability to test / maintain the parser in different configurations.

Reply:
I've added a note that behaviour will change.